### ISSUE TITLE
react components in plugins

### ISSUE THREAD

**Issue by mathisonian** [zeit/hyperterm#619]
Is there any way that we could make `require('react')` work in plugins? Passing `React` down makes sense, but it becomes a problem if you want to e.g. require a custom react component from npm that expects react to be available

**Comment by mathisonian** [comment]
Here is a more concrete example of the usecase / issue:

```js
exports.decorateTerm = (Term, { React, notify }) => {

  // this library calls require('react') internally
  const Spinner = require('react-spinner');

  return class extends React.Component {
    render () {
      const children = [];
      children.push(React.createElement(Term, Object.assign({}, this.props, { key: 'key1' })));
      children.push(React.createElement(Spinner, {key: 'key2'}));
      return React.createElement('div', {style: {width: '100%', height: '100%', position: 'relative'}}, children);
    }
  }
};
```

**Comment by rauchg** [comment]
Definitely interested in making this work.

**Comment by mathisonian** [comment]
This is coming up when trying to make a plugin depending on https://github.com/FormidableLabs/victory

![hyperchart](https://cloud.githubusercontent.com/assets/1074773/17652474/0acef1cc-624b-11e6-8c8f-a03a765377fa.gif)

Adding `react` as an explicit dependency on the plugin works for simple things but anything complex gives this warning and causes rendering bugs https://facebook.github.io/react/warnings/refs-must-have-owner.html

I'm not sure what the best approach would be to give the plugin access. Somehow expose the node_modules folder from hyperterm itself?

**Comment by TooTallNate** [comment]
Isn't this what `NODE_PATH` env var is for?

**Comment by albinekb** [comment]
This is fixed in #622 👍 
found that 'react-dom' was missing, added here: https://github.com/zeit/hyper/pull/1878
